tests/backup/backup_test.go

```diff
@@ -60,14 +54,107 @@ func TearDownBackupRestore(bkpNamespaces []string, restoreNamespaces []string) {
+	})
 	It("Backup Cluster Verification", func() {
-		Step("Check the status of pxcentral-post-install-hook pods", func() {
+		Step("Check the status of backup pods", func() {
+			dash.Info("Check the status of backup pods")
 			status := validateBackupCluster()
-			Expect(status).NotTo(Equal(false),
-				fmt.Sprintf("Backup pods are not in expected state"))
+			dash.VerifyFatal(status, true, fmt.Sprintf("Validating backup pod"))
 		})
 		//Will add CRD verification here
 	})
+	JustAfterEach(func() {
+		defer dash.TestCaseEnd()
+		log.Infof("No cleanup required for this testcase")
+	})
+})
+
+//This testcase verifies basic backup rule,backup location, cloud setting
+var _ = Describe("{BasicBackupCreateWithRules}", func() {
+	var (
+		ps       = make(map[string]map[string]string)
+		app_list = Inst().AppList
+	)
+	var contexts []*scheduler.Context
+	var CloudCredUID_list []string
+	var appContexts []*scheduler.Context
+
+	providers := getProviders()
+	JustBeforeEach(func() {
+		dash.Infof("Verifying if the pre/post rules for the required apps are present in the list or not ")
+		for i := 0; i < len(app_list); i++ {
+			if contains(post_rule_app, app_list[i]) == true {
+				if _, ok := app_parameters[app_list[i]]["post_action_list"]; ok {
+					dash.VerifyFatal(ok, true, fmt.Sprintf("Post Rule details mentioned for the apps"))
+				}
+			}
+			if contains(post_rule_app, app_list[i]) == true {
+				if _, ok := app_parameters[app_list[i]]["post_action_list"]; ok {
+					dash.VerifyFatal(ok, true, fmt.Sprintf("Pre Rule details mentioned for the apps"))
+				}
+			}
+		}
+		dash.Info("Deploy applications")
+		dash.TestCaseBegin("Backup: BasicBackupCreateWithRules", "Creating backup with Rules", "", nil)
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts = ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+		}
+	})
+	It("Basic Backup Creation", func() {
+		Step("Validate applications and get their labels", func() {
+			ValidateApplications(contexts)
+			log.Infof("Create list of pod selector for the apps deployed")
+			for _, ctx := range appContexts {
+				for _, specObj := range ctx.App.SpecList {
+					if obj, ok := specObj.(*appsapi.Deployment); ok {
+						if contains(app_list, obj.Name) == true {
+							ps[obj.Name] = obj.Spec.Template.Labels
+						}
+					} else if obj, ok := specObj.(*appsapi.StatefulSet); ok {
+						if contains(app_list, obj.Name) == true {
+							ps[obj.Name] = obj.Spec.Template.Labels
+						}
+					}
+				}
+			}
+		})
+		Step("Creating rules for backup", func() {
+			dash.Info("Creating pre rule for deployed apps")
+			pre_rule_status := createRuleForBackup("backup-pre-rule", "default", app_list, "pre", ps)
+			dash.VerifyFatal(pre_rule_status, true, fmt.Sprintf("Verifying pre rule for backup"))
```

Comment:
when there is no formatting required you can directly give the string

Reply:
Done!!